### PR TITLE
Reworks hood flipping into a component. Also adds a hood flipping ability button.

### PR DESCRIPTION
**Pull request by DimWhat**
[qol] [cleanliness]

## About the PR 
Adds an ability button for clothing that can have it's hood flipped up or down. This should make flipping hoods a lot more player friendly as people won't need to take it off to flip or unflip it. Big thanks to zjdt and pali for helping me improve the code!
image

## Why's this needed? 
You can do it very easily IRL and a lot of people don't actually know that dino PJs and hoodies actually have a hood. Also the hooded clothing subtype can be used to make future clothing items have toggleable hoods without jank copypaste code!

## Changelog

```changelog
(u)DimWhat
(*)Added an ability button for some hooded clothing.
```

### CODE REVIEW

File: code/obj/item/clothing/suits.dm [hunk 1]
```diff
@@ -28,96 +28,128 @@
...
+
+	proc/unhood()
+		if (src.hood_style)
+			src.icon_state = src.hood_style
+			usr.set_clothing_icon_dirty()
+			src.over_hair = FALSE
+			src.body_parts_covered = TORSO|ARMS
+		usr.visible_message("[usr] flips [his_or_her(usr)] [src.name]'s hood down.",\
+		"You flip your [src.name]'s hood down'.")
+		src.hooded = FALSE
```
Comment (zjdtmkhzt) [inline code/obj/item/clothing/suits.dm:64]: I would maybe advise making a hood-toggle proc instead that only toggles the hood, no message or anything, just something like src.hooded = !src.hooded and a call to UpdateIcon()?

That brings me to my next topic, maybe this whole thing should override update_icon()?

Then both the button and the attack_self proc can just call the hoodie toggle and give the person doing the toggling the message, that way you also avoid having to use usr.

You can then also just call UpdateIcon() in the New() of hoodie if necessary, to set the colored icon_state correctly. This way you avoid code duplication regarding the different icon_state settings.

Comment (DimWhat) [reply code/obj/item/clothing/suits.dm]: It has been done, thanks for the help!

**Comment by DimWhat** [comment]
Going through some componetization. Not ready yet.

File: code/datums/components/toggle_hood.dm [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+/datum/component/toggle_hood/proc/hood_icon()
+	suit.icon_state = "[src.hood_style][src.hooded ? "-up" : ""]"
+	toggle.icon_state = "hood_[src.hooded?"down":"up"]"
+
+/datum/component/toggle_hood/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_ITEM_ATTACK_SELF)
+	UnregisterSignal(parent, COMSIG_ATOM_POST_UPDATE_ICON)
+	suit.ability_buttons -= toggle
+	qdel(toggle)
```
Comment (ZeWaka) [inline code/datums/components/toggle_hood.dm:40]: toggle.suit = null?

Comment (ZeWaka) [reply code/datums/components/toggle_hood.dm:41]: no. read the above

Comment (DimWhat) [reply code/datums/components/toggle_hood.dm:41]: Sorry for the delay, I'm not certain that would even work. Toggle doesn't call upon suit anywhere.

Comment (DimWhat) [reply code/datums/components/toggle_hood.dm:40]: Do you mean just "suit = null"

Comment (ZeWaka) [reply code/datums/components/toggle_hood.dm:41]: hmm idk why i wrote that, i meant:
```suggestion
	suit = null
	qdel(toggle)
	toggle = null
```

Comment (DimWhat) [reply code/datums/components/toggle_hood.dm:41]: Huh, already done.

**Pull request merged**